src/main/java/io/github/hildi/can/service/JavaSerializationService.java

```diff
@@ -0,0 +1,239 @@
+            userNickName = line[1];
+        }
+    }
+
+    private void getUserFirstName(String param, String[] line) {
+        if (param.equals("firstName".toLowerCase().trim())) {
+            firstName = line[1].trim();
+        }
+    }
+
+    private void getUserLastName(String param, String[] line) {
+        if (param.equals("lastName".toLowerCase().trim())) {
+            lastName = line[1].trim();
+        }
+    }
+
+    private void getUserEmail(String param, String[] line) {
+        if (param.equals("email".toLowerCase().trim())) {
+            userEmail = line[1].toLowerCase().trim();
+        }
+    }
+
+    private void getUserPermissions(String param, String[] line) {
+        if (param.equals("permissions".toLowerCase().trim())) {
+            String trimLine = line[1].trim();
+            String[] permissionsSplit = trimLine.split(" , ");
+            userPermissions = new ArrayList<>(Arrays.asList(permissionsSplit));
+        }
+    }
+
+    private void getUserAttributes(String param, String[] line) {
+        if (param.equals("attributes".toLowerCase().trim())) {
+            userAttributes = new LinkedHashMap<>();
+            // country: ukraine; city: kharkiv
+            String[] attributesSplit = line[1].trim().split("; ");
+            // country: ukraine
+            String[] countrySplit = attributesSplit[0].trim().split(": ");
+            // city: kharkiv
+            String[] citySplit = attributesSplit[1].trim().split(": ");
+            // country
+            // ukraine
+            userAttributes.put(countrySplit[0], countrySplit[1]);
+            // city
+            // kharkiv
+            userAttributes.put(citySplit[0], citySplit[1]);
+        }
+    }
+
+    private void getUserCreatedData(String param, String[] line) {
+        if (param.equals("createdAt".toLowerCase().trim())) {
+            String[] split = line[1].split("T");
+            String[] splitDate = split[0].split("-");
+            String[] splitTime = split[1].split(":");
+
+            Calendar calendar = Calendar.getInstance();
+            calendar.set(Calendar.YEAR, Integer.parseInt(splitDate[0]));
+            calendar.set(Calendar.MONTH, Integer.parseInt(splitDate[1]) - 1);
+            calendar.set(Calendar.DAY_OF_MONTH, Integer.parseInt(splitDate[2]));
+            calendar.set(Calendar.HOUR_OF_DAY, Integer.parseInt(splitTime[0]));
+            calendar.set(Calendar.MINUTE, Integer.parseInt(splitTime[1]));
+            calendar.set(Calendar.SECOND, Integer.parseInt(splitTime[2]));
+            createdAt = calendar.getTime();
+        }
+    }
+
+    private static void assertFileExtensionIsCorrect(File file) {
+        String name = file.getName();
+        if (!name.endsWith(".properties")){
+            throw new InvalidFileException("Failed to serialize data, reason: invalid file extension " + file.getName());
+        }
+    }
+
+    private static void assertFileWritable(File file) {
```

Comment:
You already have all these methods in another class. 
Consider extracting them into some utility class and reusing them here.